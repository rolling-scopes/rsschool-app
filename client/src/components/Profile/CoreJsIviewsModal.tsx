import * as React from 'react';
// import isEqual from 'lodash/isEqual'
import { CoreJsInterviewsData } from './CoreJsIviewsCard';
import { Modal, Table, Typography, Tag } from 'antd';

const { Text } = Typography;

type Props = {
  stats: CoreJsInterviewsData;
  interviewIndex: number;
  isVisible: boolean;
  onHide: () => void;
};

class CoreJsIviewsModal extends React.PureComponent<Props> {
  render() {
    const { stats, isVisible, onHide, interviewIndex } = this.props;
    const { courseFullName, interviews } = stats;
    const { score, comment, answers, interviewer } = interviews[interviewIndex];

    return (
      <Modal
        title={`${courseFullName} CoreJS Interview Feedback`}
        visible={isVisible}
        onCancel={onHide}
        footer={null}
        width={'80%'}
      >
        <p style={{ marginBottom: 5 }}>
          Score: <Text mark>{score}</Text>
        </p>
        <p style={{ marginBottom: 20 }}>
          Interviewer: <a href={`/profile?githubId=${interviewer.githubId}`}>{interviewer.name}</a>
        </p>
        {comment && (
          <p style={{ marginBottom: 20 }}>
            <Text strong>Comment: </Text>
            {comment}
          </p>
        )}
        <Table
          data-testid="profile-corejs-iviews-modal-table"
          dataSource={answers}
          size="small"
          rowKey="questionId"
          pagination={false}
          columns={[
            {
              dataIndex: 'questionText',
              ellipsis: true,
            },
            {
              dataIndex: 'answer',
              render: answer =>
                answer === true ? <Tag color="green">Yes</Tag> : answer === false ? <Tag color="red">No</Tag> : answer,
            },
          ]}
        />
      </Modal>
    );
  }
}

export default CoreJsIviewsModal;
