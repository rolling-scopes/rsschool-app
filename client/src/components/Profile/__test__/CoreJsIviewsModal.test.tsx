import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CoreJsIviewsModal from '../CoreJsIviewsModal';

describe('CoreJsIviewsModal', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(
      <CoreJsIviewsModal
        stats={{
          courseFullName: 'rs-2019',
          courseName: 'rs-2019',
          locationName: 'minsk',
          interviews: [
            {
              answers: [
                {
                  answer: 'true',
                  questionText: 'test',
                  questionId: 'test',
                },
                {
                  answer: 'false',
                  questionText: 'test',
                  questionId: 'test',
                },
              ],
              interviewer: {
                name: 'Dzmitry Petrov',
                githubId: 'dima',
              },
              comment: 'test',
              score: 4,
            },
          ],
        }}
        isVisible={true}
        onHide={jest.fn()}
        interviewIndex={0}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
