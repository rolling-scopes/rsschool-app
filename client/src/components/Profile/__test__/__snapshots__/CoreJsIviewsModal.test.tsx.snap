// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreJsIviewsModal Should render correctly 1`] = `
<Modal
  confirmLoading={false}
  footer={null}
  okType="primary"
  onCancel={[MockFunction]}
  title="rs-2019 CoreJS Interview Feedback"
  visible={true}
  width="80%"
>
  <p
    style={
      Object {
        "marginBottom": 5,
      }
    }
  >
    Score: 
    <Text
      mark={true}
    >
      4
    </Text>
  </p>
  <p
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    Interviewer: 
    <a
      href="/profile?githubId=dima"
    >
      Dzmitry Petrov
    </a>
  </p>
  <p
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      strong={true}
    >
      Comment: 
    </Text>
    test
  </p>
  <ForwardRef(InternalTable)
    columns={
      Array [
        Object {
          "dataIndex": "questionText",
          "ellipsis": true,
        },
        Object {
          "dataIndex": "answer",
          "render": [Function],
        },
      ]
    }
    dataSource={
      Array [
        Object {
          "answer": "true",
          "questionId": "test",
          "questionText": "test",
        },
        Object {
          "answer": "false",
          "questionId": "test",
          "questionText": "test",
        },
      ]
    }
    pagination={false}
    rowKey="questionId"
    size="small"
  />
</Modal>
`;
